lib/rust/parser/src/macros/resolver.rs

```diff
@@ -421,9 +421,14 @@ impl<'s> ResolverState<'s> {
                     Self::move_to_next_segment(&mut current_macro.matched_macro_def, subsegments);
                 mem::swap(&mut new_match_tree, &mut current_macro.possible_next_segments);
                 return Step::StartSegment(token);
-            } else if let Some(popped) = self.pop_macro_stack_if_reserved(repr) {
-                self.resolve(popped);
-                return Step::MacroStackPop(token.into());
+            } else {
+                match self.pop_macro_stack_if_reserved(repr) {
+                    Some(popped) => {
+                        self.resolve(popped);
+                        return Step::MacroStackPop(token.into());
+                    }
+                    _ => {}
+                }
             }
```

Comment:
I don't feel it is better now (`if let` expresses the thing more concisely), but perhaps it's a matter of taste.